lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,379 @@
+
+- name: copy file to vm
+  vmware_guest_file_operation:
+    hostname: myVSphere
+    username: myUsername
+    password: mySecret
+    datacenter: myDatacenter
+    validate_certs: True
+    folder: /vm
+    vm_id: NameOfVM
+    vm_username: root
+    vm_password: superSecret
+    copy:
+        src: "files/test.zip"
+        dest: "/root/test.zip"
+        overwrite: False
+  delegate_to: localhost
+
+- name: fetch file from vm
+  vmware_guest_file_operation:
+    hostname: myVSphere
+    username: myUsername
+    password: mySecret
+    datacenter: myDatacenter
+    validate_certs: True
+    folder: /vm
+    vm_id: NameOfVM
+    vm_username: root
+    vm_password: superSecret
+    fetch:
+        src: "/root/test.zip"
+        dest: "files/test.zip"
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils import urls
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.module_utils.vmware import (connect_to_api, find_cluster_by_name, find_datacenter_by_name,
+                                         find_vm_by_id, HAS_PYVMOMI, vmware_argument_spec, PyVmomi)
+
+
+def directory(module, content, vm):
+    operations_permit = ['create', 'delete']
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+
+    if "path" not in module.params["directory"]:
+        module.fail_json(msg="directory.path is mandatory")
+    if "operation" not in module.params["directory"]:
+        module.fail_json(msg="directory.operation is mandatory")
+    if "recurse" not in module.params["directory"]:
+        recurse = False
+    else:
+        recurse = bool(module.params['directory']['recurse'])
+    operation = module.params["directory"]['operation']
+    path = module.params["directory"]['path']
+    if operation not in operations_permit:
+        module.fail_json(msg="directory.operation valid values are create, delete")
+
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+    if operation == "create":
+        try:
+            file_manager.MakeDirectoryInGuest(vm=vm, auth=creds, directoryPath=path,
+                                              createParentDirectories=recurse)
+        except vim.fault.FileAlreadyExists:
+            module.fail_json(msg="Directory %s already exist" % path)
+        except vim.fault.GuestPermissionDenied:
+            module.fail_json(msg="Permission denied for path %s" % path)
+        except vim.fault.InvalidGuestLogin:
+            module.fail_json(msg="Invalid guest login for user %s" % vm_username)
+        # other exceptions
+        except Exception as e:
+            module.fail_json(msg="Failed to Create directory into Vm VMware exception:%s" % e)
+
+    if operation == "delete":
```

Comment:
Preferably, we first check if the directory exists, if not, we just return a `changed=False` proper exit